runtime/src/bank.rs

```diff
@@ -899,9 +897,7 @@ impl Bank {
             fee_rate_governor: fields.fee_rate_governor,
             collected_rent: AtomicU64::new(fields.collected_rent),
             // clone()-ing is needed to consider a gated behavior in rent_collector
-            rent_collector: fields
-                .rent_collector
-                .clone_with_epoch(fields.epoch, genesis_config.cluster_type),
+            rent_collector: fields.rent_collector.clone_with_epoch(fields.epoch),
```

Comment:
change of plumbing by removing `RentCollecotr::cluster_type` wasn't avoidable exactly because of unavailability of feature set of new bank even before constructing the very first bank from the snapshot..